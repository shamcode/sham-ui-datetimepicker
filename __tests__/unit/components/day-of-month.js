import DayOfMonth  from '../../../src/components/day-of-month.sfc';
import { onclick } from 'sham-ui-directives';
import renderer from 'sham-ui-test-helpers';

it( 'renders correctly', () => {
    const meta = renderer( DayOfMonth, {
        today: new Date( 2020, 3, 25, 15, 3, 30 ),
        value: new Date( 2015, 1, 23, 13, 2, 17 ),
        date: new Date( 2015, 1, 23, 13, 2, 17 )
    }, {
        directives: {
            onclick
        }
    } );
    expect( meta.toJSON() ).toMatchSnapshot();
} );


it( 'changeMonth', () => {
    const meta = renderer( DayOfMonth, {
        today: new Date( 2020, 3, 25, 15, 3, 30 ),
        value: new Date( 2015, 1, 23, 13, 2, 17 ),
        date: new Date( 2015, 1, 23, 13, 2, 17 )
    }, {
        directives: {
            onclick
        }
    } );
    meta.ctx.container.querySelector( 'thead tr:first-child th:last-child' ).click();
    expect( meta.toJSON() ).toMatchSnapshot();
    meta.ctx.container.querySelector( 'thead tr:first-child th:first-child' ).click();
    expect( meta.toJSON() ).toMatchSnapshot();
} );
