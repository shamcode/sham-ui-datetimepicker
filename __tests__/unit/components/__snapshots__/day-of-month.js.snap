// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changeMonth 1`] = `
Object {
  "Constructor": "DayOfMonth",
  "Options": Object {
    "ID": "component",
    "changeDate": [Function],
    "classForDate": [Function],
    "date": 2015-03-23T07:02:17.000Z,
    "daysOfWeek": Array [
      "Mon",
      "Tue",
      "Wed",
      "Thu",
      "Fri",
      "Sat",
      "Sun",
    ],
    "directives": Object {
      "onclick": [Function],
    },
    "isDateSelectable": [Function],
    "nextSymbol": "▸",
    "prevSymbol": "◂",
    "showTime": true,
    "today": 2020-04-25T08:03:30.000Z,
    "value": 2015-02-23T07:02:17.000Z,
  },
  "html": "
<thead>
  <tr>
    <th><span>◂</span></th>
    <th colspan=\\"5\\" class=\\"picker-switch\\">2015 ├F3: M03┤</th>
    <th><span>▸</span></th>
  </tr>
  <tr>
    <th>Mon</th>
    <th>Tue</th>
    <th>Wed</th>
    <th>Thu</th>
    <th>Fri</th>
    <th>Sat</th>
    <th>Sun</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td class=\\"day other-month active\\">23</td>
    <td class=\\"day other-month\\">24</td>
    <td class=\\"day other-month\\">25</td>
    <td class=\\"day other-month\\">26</td>
    <td class=\\"day other-month\\">27</td>
    <td class=\\"day other-month\\">28</td>
    <td class=\\"day\\">1</td>
  </tr>
  <tr>
    <td class=\\"day\\">2</td>
    <td class=\\"day\\">3</td>
    <td class=\\"day\\">4</td>
    <td class=\\"day\\">5</td>
    <td class=\\"day\\">6</td>
    <td class=\\"day\\">7</td>
    <td class=\\"day\\">8</td>
  </tr>
  <tr>
    <td class=\\"day\\">9</td>
    <td class=\\"day\\">10</td>
    <td class=\\"day\\">11</td>
    <td class=\\"day\\">12</td>
    <td class=\\"day\\">13</td>
    <td class=\\"day\\">14</td>
    <td class=\\"day\\">15</td>
  </tr>
  <tr>
    <td class=\\"day\\">16</td>
    <td class=\\"day\\">17</td>
    <td class=\\"day\\">18</td>
    <td class=\\"day\\">19</td>
    <td class=\\"day\\">20</td>
    <td class=\\"day\\">21</td>
    <td class=\\"day\\">22</td>
  </tr>
  <tr>
    <td class=\\"day\\">23</td>
    <td class=\\"day\\">24</td>
    <td class=\\"day\\">25</td>
    <td class=\\"day\\">26</td>
    <td class=\\"day\\">27</td>
    <td class=\\"day\\">28</td>
    <td class=\\"day\\">29</td>
  </tr>
  <tr>
    <td class=\\"day\\">30</td>
    <td class=\\"day\\">31</td>
    <td class=\\"day other-month\\">1</td>
    <td class=\\"day other-month\\">2</td>
    <td class=\\"day other-month\\">3</td>
    <td class=\\"day other-month\\">4</td>
    <td class=\\"day other-month\\">5</td>
  </tr>
</tbody>
<tfoot>
  <tr>
    <th colspan=\\"7\\">13:02:17</th>
  </tr>
</tfoot>
<!--if-->
",
}
`;

exports[`changeMonth 2`] = `
Object {
  "Constructor": "DayOfMonth",
  "Options": Object {
    "ID": "component",
    "changeDate": [Function],
    "classForDate": [Function],
    "date": 2015-02-23T07:02:17.000Z,
    "daysOfWeek": Array [
      "Mon",
      "Tue",
      "Wed",
      "Thu",
      "Fri",
      "Sat",
      "Sun",
    ],
    "directives": Object {
      "onclick": [Function],
    },
    "isDateSelectable": [Function],
    "nextSymbol": "▸",
    "prevSymbol": "◂",
    "showTime": true,
    "today": 2020-04-25T08:03:30.000Z,
    "value": 2015-02-23T07:02:17.000Z,
  },
  "html": "
<thead>
  <tr>
    <th><span>◂</span></th>
    <th colspan=\\"5\\" class=\\"picker-switch\\">2015 ├F3: M02┤</th>
    <th><span>▸</span></th>
  </tr>
  <tr>
    <th>Mon</th>
    <th>Tue</th>
    <th>Wed</th>
    <th>Thu</th>
    <th>Fri</th>
    <th>Sat</th>
    <th>Sun</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td class=\\"day other-month\\">26</td>
    <td class=\\"day other-month\\">27</td>
    <td class=\\"day other-month\\">28</td>
    <td class=\\"day other-month\\">29</td>
    <td class=\\"day other-month\\">30</td>
    <td class=\\"day other-month\\">31</td>
    <td class=\\"day\\">1</td>
  </tr>
  <tr>
    <td class=\\"day\\">2</td>
    <td class=\\"day\\">3</td>
    <td class=\\"day\\">4</td>
    <td class=\\"day\\">5</td>
    <td class=\\"day\\">6</td>
    <td class=\\"day\\">7</td>
    <td class=\\"day\\">8</td>
  </tr>
  <tr>
    <td class=\\"day\\">9</td>
    <td class=\\"day\\">10</td>
    <td class=\\"day\\">11</td>
    <td class=\\"day\\">12</td>
    <td class=\\"day\\">13</td>
    <td class=\\"day\\">14</td>
    <td class=\\"day\\">15</td>
  </tr>
  <tr>
    <td class=\\"day\\">16</td>
    <td class=\\"day\\">17</td>
    <td class=\\"day\\">18</td>
    <td class=\\"day\\">19</td>
    <td class=\\"day\\">20</td>
    <td class=\\"day\\">21</td>
    <td class=\\"day\\">22</td>
  </tr>
  <tr>
    <td class=\\"day active\\">23</td>
    <td class=\\"day\\">24</td>
    <td class=\\"day\\">25</td>
    <td class=\\"day\\">26</td>
    <td class=\\"day\\">27</td>
    <td class=\\"day\\">28</td>
    <td class=\\"day other-month\\">1</td>
  </tr>
  <tr>
    <td class=\\"day other-month\\">2</td>
    <td class=\\"day other-month\\">3</td>
    <td class=\\"day other-month\\">4</td>
    <td class=\\"day other-month\\">5</td>
    <td class=\\"day other-month\\">6</td>
    <td class=\\"day other-month\\">7</td>
    <td class=\\"day other-month\\">8</td>
  </tr>
</tbody>
<tfoot>
  <tr>
    <th colspan=\\"7\\">13:02:17</th>
  </tr>
</tfoot>
<!--if-->
",
}
`;

exports[`renders correctly 1`] = `
Object {
  "Constructor": "DayOfMonth",
  "Options": Object {
    "ID": "component",
    "changeDate": [Function],
    "classForDate": [Function],
    "date": 2015-02-23T07:02:17.000Z,
    "daysOfWeek": Array [
      "Mon",
      "Tue",
      "Wed",
      "Thu",
      "Fri",
      "Sat",
      "Sun",
    ],
    "directives": Object {
      "onclick": [Function],
    },
    "isDateSelectable": [Function],
    "nextSymbol": "▸",
    "prevSymbol": "◂",
    "showTime": true,
    "today": 2020-04-25T08:03:30.000Z,
    "value": 2015-02-23T07:02:17.000Z,
  },
  "html": "
<thead>
  <tr>
    <th><span>◂</span></th>
    <th colspan=\\"5\\" class=\\"picker-switch\\">2015 ├F3: M02┤</th>
    <th><span>▸</span></th>
  </tr>
  <tr>
    <th>Mon</th>
    <th>Tue</th>
    <th>Wed</th>
    <th>Thu</th>
    <th>Fri</th>
    <th>Sat</th>
    <th>Sun</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td class=\\"day other-month\\">26</td>
    <td class=\\"day other-month\\">27</td>
    <td class=\\"day other-month\\">28</td>
    <td class=\\"day other-month\\">29</td>
    <td class=\\"day other-month\\">30</td>
    <td class=\\"day other-month\\">31</td>
    <td class=\\"day\\">1</td>
  </tr>
  <tr>
    <td class=\\"day\\">2</td>
    <td class=\\"day\\">3</td>
    <td class=\\"day\\">4</td>
    <td class=\\"day\\">5</td>
    <td class=\\"day\\">6</td>
    <td class=\\"day\\">7</td>
    <td class=\\"day\\">8</td>
  </tr>
  <tr>
    <td class=\\"day\\">9</td>
    <td class=\\"day\\">10</td>
    <td class=\\"day\\">11</td>
    <td class=\\"day\\">12</td>
    <td class=\\"day\\">13</td>
    <td class=\\"day\\">14</td>
    <td class=\\"day\\">15</td>
  </tr>
  <tr>
    <td class=\\"day\\">16</td>
    <td class=\\"day\\">17</td>
    <td class=\\"day\\">18</td>
    <td class=\\"day\\">19</td>
    <td class=\\"day\\">20</td>
    <td class=\\"day\\">21</td>
    <td class=\\"day\\">22</td>
  </tr>
  <tr>
    <td class=\\"day active\\">23</td>
    <td class=\\"day\\">24</td>
    <td class=\\"day\\">25</td>
    <td class=\\"day\\">26</td>
    <td class=\\"day\\">27</td>
    <td class=\\"day\\">28</td>
    <td class=\\"day other-month\\">1</td>
  </tr>
  <tr>
    <td class=\\"day other-month\\">2</td>
    <td class=\\"day other-month\\">3</td>
    <td class=\\"day other-month\\">4</td>
    <td class=\\"day other-month\\">5</td>
    <td class=\\"day other-month\\">6</td>
    <td class=\\"day other-month\\">7</td>
    <td class=\\"day other-month\\">8</td>
  </tr>
</tbody>
<tfoot>
  <tr>
    <th colspan=\\"7\\">13:02:17</th>
  </tr>
</tfoot>
<!--if-->
",
}
`;
